src/components/screens/editor/editor-screen.tsx

```diff
@@ -8,64 +8,60 @@ import Api from "api";
 import { NavigationTabContainer } from "styled/layouts/navigations";
 import NavigationTab from "components/layouts/navigations/navigation-tab";
 import { Metaform, MetaformVersion, MetaformVersionType } from "generated/client";
-import { AdminFormListStack, AdminFormTypographyField, VersionListHeader } from "styled/react-components/react-components";
-import { DataGrid, GridColDef } from "@mui/x-data-grid";
 import { NewUserButton } from "styled/layouts/admin-layout";
-import moment from "moment";
 import EditorScreenDrawer from "./editor-screen-drawer";
 import { useNavigate } from "react-router-dom";
-
-/**
- * Interface for single Metaform row
- */
-interface Row {
-  metaform: Metaform;
-  versions: MetaformVersion[];
-}
+import GenericLoaderWrapper from "components/generic/generic-loader";
+import EditorScreenTable from "./editor-table";
 
 /**
  * Editor screen component
  */
 const EditorScreen: React.FC = () => {
+  const currentPath = window.location.pathname;
   const errorContext = useContext(ErrorContext);
   const navigate = useNavigate();
 
   const apiClient = useApiClient(Api.getApiClient);
   const { metaformsApi, versionsApi } = apiClient;
 
-  const [ metaformRows, setMetaformRows ] = useState<Row[]>([]);
+  const [ metaforms, setMetaforms ] = useState<Metaform[]>([]);
+  const [ metaformVersions, setMetaformVersions ] = useState<MetaformVersion[]>([]);
+  const [ selectedId, setSelectedId ] = useState<string | undefined>();
```

Comment:
Could be more describing, deletableId etc.

Reply:
If this is used only for deleting, you should pass id as function parameter, not through state.

Reply:
Done.